### PR TITLE
Add missing id, name fields to json schema docs

### PR DESCRIPTION
**Pull request by hellosmithy**
Added the optional `id` and `name` fields which are documented on the FINOS site but currently missing from `OrderList`, `InstrumentList`, `ContactList` and `Contact` (name only) json schema docs.

See:

- [OrderList](https://fdc3.finos.org/docs/context/ref/OrderList)
- [InstrumentList](https://fdc3.finos.org/docs/context/ref/InstrumentList)
- [ContactList](https://fdc3.finos.org/docs/context/ref/ContactList)
- [Contact](https://fdc3.finos.org/docs/context/ref/Contact)

### CODE REVIEW

**Comment by kriswest** [comment]
Hi @hellosmithy, technically the `id` and `name` properties are 'inherited' from the base Context schema, which is composed with each of these via `allOf`: https://github.com/finos/FDC3/blob/13ccdd2b7b57aa81721ee136e54f42e27e65cf8c/schemas/context/contact.schema.json#L41C28-L41C28

which end up in the TypeScript types generated from the resulting composed schema:
https://github.com/finos/FDC3/blob/dcd98c2111d4d815ddf7d5cd3f4d91c536be414a/src/context/ContextTypes.ts#L1247-L1266 

However, it's not really inheritance (composition of schemas is more like layering filters over one another) and there is no problem defining them in multiple schemas that get composed together. You'll note we use the `BaseContext` without docs, allowing the downstream schemas to add their own without doubling up - which your PR does.

Finally, a note for the maintainers, the preview for PR #1151 (generate Context docs from schemas) shows that the docs generation script is not picking up the composed schema refs and rendering the Context schema properties: https://issue-1068--exquisite-otter-12d363.netlify.app/docs/context/schemas/Contact
We can either fix that (not trivial) or continue down this path and document the id and name properties where they make sense or are more tightly defined (e.g. specific id subfields). This latter option is simpler and adds more value to the docs / avoids rendering `name` and `id` in context docs where they are not so relevant.

**Review by kriswest — DISMISSED** [review]
LGTM - although we should review if there are more schemas that need a similar change

**Comment by kriswest** [comment]
@hellosmithy Once you've got a CLA manger setup, please click on the [Please click here to be authorized](https://api.easycla.lfx.linuxfoundation.org/v2/repository-provider/github/sign/13399143/121826936/1233/#/?version=2) link above to have EasyCLA check it and add you to its database of approved contributors, which will allow us to merge this once reviewed.

**Review by hughtroeger — COMMENTED** [review]
As @kriswest mentioned there are a few other other context types missing id and/or name (see my comment here https://github.com/finos/FDC3/pull/1151#pullrequestreview-2086687762 )

- Country is missing name
- Organization is missing name
- Portfolio is missing name and id
- Position is missing name and id
- TradeList is missing name and id

**Comment by hellosmithy** [comment]
> technically the `id` and `name` properties are 'inherited' from the base Context schema, which is composed with each of these via `allOf`

Ah yes, I knew about the composition but didn't realise it included `id` and `name`. For some reason the `json-schema-to-typescript` npm package I'm using to generate some types also doesn't appear to pick these up which is what brought my attention to it.

**Comment by hellosmithy** [comment]
Slight tangent, but are `type` fields supposed to be case sensitive or insensitive?
[TimeRange](https://fdc3.finos.org/schemas/2.1/context/timerange.schema.json) has camel case in the example, but lowercase in the schema

**Comment by kriswest** [comment]
Ah yes, I knew about the composition but didn't realise it included id and name. For some reason the json-schema-to-typescript npm package I'm using to generate some types also doesn't appear to pick these up which is what brought my attention to it.
If I remember rightly, json-schema-to-typescript doesn't retrieve references ($ref). You can can do that yourself by just replacing the $ref element with the referenced schema. https://www.npmjs.com/package/json-schema-resolver I think can do that for you as well.

In the FDC3 repo we use quicktype (which possibly combines these two libs under-the-hood) to generate typescript types from the schemas.

K

[cid:7e2c8f9f-afac-425f-b435-2a0daf5ca47f]


Kris West

[cid:b7dc33e4-7c13-4156-bfa9-fb3fa3a56724]<https://www.linkedin.com/in/kriswest/>  Director, Consulting Services
+44 7715104292
***@***.******@***.***>


________________________________

**Comment by kriswest** [comment]
@hellosmithy could you run `npm run typegen` or `npm run build` for me and check in the changes it makes? This will regenerate the typescript files produced from the schema and ensure they make it into the diff. 

I should probably have this added to the pre-commit hook so that it's automated!

File: src/context/ContextTypes.ts
Comment (hellosmithy) [inline src/context/ContextTypes.ts:789]: For some reason these generated comments are leaking across the types in a couple of places. It's not immediately clear to me why

File: src/context/ContextTypes.ts
Comment (hellosmithy) [inline src/context/ContextTypes.ts:1131]: Here again

File: src/context/ContextTypes.ts
Comment (kriswest) [reply src/context/ContextTypes.ts:789]: QuickType has a really annoying habit of combining similar entities based on heuristics. You can read more about it here: http://blog.quicktype.io/under-the-hood/  and can see others trying to deal with it here: https://github.com/glideapps/quicktype/issues/1678

Generally, it doesn't cause problems as the types are usually correct. However, it does mess with these docs. I'll take a look as you can sometimes getting it to stop doing it by altering or setting the `title` element...

**Comment by kriswest** [comment]
@hellosmithy I was able to improve the naming of the generated object, but not stop it from doubling up on the comments. What's happening is that quicktype is combining two schemas which a field could be `oneOf` or `anyOf` to produce a type that would match any of the allowed types. Hence, its got two comments as it could be either. 

In ChatSearchCriteria  each criteria could be an instrument, a contact, an organisation or a free text string. I was at least able to get the object named properly (to `SearchCriteria`) and added a description to fdc3.instrument's `name` property so at least we're consistent:
https://github.com/finos/FDC3/blob/5e3eee3b4a645eae14b2c85d9ae4daac1dc52158/src/context/ContextTypes.ts#L791-L798

**Comment by kriswest** [comment]
@hellosmithy any joy on the CLA front? If there's anything we or FINOS can do to help let us know!

**Comment by hellosmithy** [comment]
> @hellosmithy any joy on the CLA front? If there's anything we or FINOS can do to help let us know!

@kriswest not yet other than that it's going through the various channels - should I close this PR until we have an update?

**Comment by kriswest** [comment]
No problem leaving it open. If its likely to take more than a few weeks we'll recreate the changes in a new PR to close out the issue.

**Comment by kriswest** [comment]
@hellosmithy any update on that approval/CLA? We're talking about prereleases for review and it would be good to get this in...

**Comment by hellosmithy** [comment]
> @hellosmithy any update on that approval/CLA? We're talking about prereleases for review and it would be good to get this in...

@kriswest it's been raised to all the people it needs to be, but we don't yet have an ETA I'm afraid.
Since these are fairly trivial changes it might be best for someone else to re-submit the PR and I can close this one. Wdyt?

**Comment by kriswest** [comment]
superseded by #1360

**Pull request closed without merging**
